# [실습] 단변량 분석

# 1.비즈니스 시나리오


- 고객사는 카시트 전문 판매 회사입니다.
    - 최근, 매출 그래프가 조금씩 아래로 향하기 시작했습니다. 회사의 여러 부서에서는 다양한 원인을 제시했지만, 그 원인들이 정말 데이터에 기반한 것인지는 여전히 의문입니다.

    - 부서별 문제 진단:
        - 가격 경쟁력 약화: 경쟁사와의 가격 경쟁에서 밀리고 있음
        - 광고비 집행 비효율: 광고비는 쓰고 있지만, 정말 적절하게 집행되지 않음
        - 국내 시장 편중: 해외 시장 개척은 뒷전이고, 국내 시장에만 집중됨
        - 지역 구매력 미파악: 지역별 구매력과 특성을 제대로 분석하지 못하고 있음

- 이 문제를 해결하기 위해 DX 프로젝트 팀이 빌딩되었습니다.
    - 여러분은 이 팀에서 데이터 분석을 책임지는 DX 컨설턴트로 합류했습니다.
    - 드디어 갈고 닦은 실력을 발휘할 기회가 주어졌습니다.
    - 자, 이제 단변량 분석을 통해 비즈니스 인사이트를 도출해 봅시다!
----
* 반복되는 코드는 함수로 만듭시다!
    * 함수를 만드는 순서
        * 먼저 절차대로 코드를 작성해서 실행한다.
        * 실행된 코드(오류가 없는 코드)를 가져다 함수로 구성한다.
            * 입력과 출력 결정

# 2.환경준비

## 2.1. 라이브러리 불러오기

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 2.2. 데이터 불러오기 : 카시트 판매량 데이터

|	변수명	|	설명	|	구분	|
|	----	|	----	|	----	|
|	Sales 	|	 각 지역 판매량(단위 : 1000개)	|	Target	|
|	CompPrice 	|	 경쟁사 가격(단위 : 달러)	|	feature	|
|	Income 	|	 지역 평균 소득(단위 : 1000달러)	|	feature	|
|	Advertising 	|	 각 지역, 회사의 광고 예산(단위 : 1000달러)	|	feature	|
|	Population 	|	 지역 인구수(단위 : 1000명)	|	feature	|
|	Price 	|	 자사 지역별 판매가격	|	feature	|
|	ShelveLoc 	|	 진열상태	|	feature	|
|	Age 	|	 지역 인구의 평균 연령	|	feature	|
|	Urban 	|	 도심 지역 여부(Yes,No)	|	feature	|
|	US 	|	 매장이 미국에 있는지 여부(Yes,No)	|	feature	|


In [2]:
path = './data/Carseats.csv'

car = pd.read_csv(path)

car.head()

,Sales,CompPrice,Income,Advertising,Population,Price,ShelveLoc,Age,Urban,US
0,9.50,138,73,11,276,120,Bad,42,Yes,Yes
1,11.22,111,48,16,260,83,Good,65,Yes,Yes
2,10.06,113,35,10,269,80,Medium,59,Yes,Yes
3,7.40,117,100,4,466,97,Medium,55,Yes,Yes
4,4.15,141,64,3,340,128,Bad,38,Yes,No


In [3]:
car.describe()

,Sales,CompPrice,Income,Advertising,Population,Price,Age
count,400.000000,400.000000,400.000000,400.000000,400.000000,400.000000,400.000000
mean,7.496325,124.975000,68.657500,6.635000,264.840000,115.795000,53.322500
std,2.824115,15.334512,27.986037,6.650364,147.376436,23.676664,16.200297
min,0.000000,77.000000,21.000000,0.000000,10.000000,24.000000,25.000000
25%,5.390000,115.000000,42.750000,0.000000,139.000000,100.000000,39.750000
50%,7.490000,125.000000,69.000000,5.000000,272.000000,117.000000,54.500000
75%,9.320000,135.000000,91.000000,12.000000,398.500000,131.000000,66.000000
max,16.270000,175.000000,120.000000,29.000000,509.000000,191.000000,80.000000


In [4]:
car.info()

<class 'pandas.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sales        400 non-null    float64
 1   CompPrice    400 non-null    int64  
 2   Income       400 non-null    int64  
 3   Advertising  400 non-null    int64  
 4   Population   400 non-null    int64  
 5   Price        400 non-null    int64  
 6   ShelveLoc    400 non-null    str    
 7   Age          400 non-null    int64  
 8   Urban        400 non-null    str    
 9   US           400 non-null    str    
dtypes: float64(1), int64(6), str(3)
memory usage: 31.4 KB


# 3.단변량 분석 : 숫자형 변수

In [ ]:
def eda_num(df, col, bins=20):
    # 기초통계량
    display(df[col].describe().to_frame().T)
    print(f'왜도(skewness): {df[col].skew():.4f}  |  첨도(kurtosis): {df[col].kurt():.4f}')

    # 시각화
    fig, axes = plt.subplots(1, 2, figsize=(12, 4))

    df[col].hist(ax=axes[0], bins=bins, edgecolor='k', color='steelblue', alpha=0.7)
    axes[0].axvline(df[col].mean(),   color='red',   linestyle='--', label=f'mean={df[col].mean():.2f}')
    axes[0].axvline(df[col].median(), color='green', linestyle='--', label=f'median={df[col].median():.2f}')
    axes[0].set_title(f'{col} Histogram')
    axes[0].legend()

    axes[1].boxplot(df[col], vert=True)
    axes[1].set_title(f'{col} Boxplot')
    axes[1].set_ylabel(col)

    plt.suptitle(col, fontsize=13, fontweight='bold')
    plt.tight_layout()
    plt.show()

## 3.1. Sales (Target)

In [ ]:
eda_num(car, 'Sales')

1) 변수의 비즈니스 의미

* 각 지역 판매량(단위 : 1000개)
* Target

2) 기초통계량(수치화)

In [ ]:
car['Sales'].describe()

In [ ]:
print(f'왜도(skewness): {car["Sales"].skew():.4f}')
print(f'첨도(kurtosis): {car["Sales"].kurt():.4f}')

Q1 = car['Sales'].quantile(0.25)
Q3 = car['Sales'].quantile(0.75)
IQR = Q3 - Q1
print(f'\nIQR: {IQR:.4f}')
print(f'하한(Q1 - 1.5*IQR): {Q1 - 1.5*IQR:.4f}')
print(f'상한(Q3 + 1.5*IQR): {Q3 + 1.5*IQR:.4f}')

3) 분포 확인(시각화)

In [ ]:
fig, axes = plt.subplots(1, 2, figsize=(12, 4))

car['Sales'].hist(ax=axes[0], bins=20, edgecolor='k', color='steelblue', alpha=0.7)
axes[0].axvline(car['Sales'].mean(),   color='red',   linestyle='--', label=f"mean={car['Sales'].mean():.2f}")
axes[0].axvline(car['Sales'].median(), color='green', linestyle='--', label=f"median={car['Sales'].median():.2f}")
axes[0].set_title('Sales Histogram')
axes[0].set_xlabel('Sales (1000개)')
axes[0].set_ylabel('Frequency')
axes[0].legend()

axes[1].boxplot(car['Sales'], vert=True)
axes[1].set_title('Sales Boxplot')
axes[1].set_ylabel('Sales (1000개)')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 평균 판매량은 약 **7,496개**(7.496천 개)이며, 중앙값(7.490)과 거의 같아 분포가 비교적 대칭적이다.
- 왜도가 0에 가깝고 히스토그램이 정규분포에 가까운 종 모양을 보인다.
- 판매량 0~2천 개 수준의 하위 지역이 일부 존재하며, 이 지역들의 특성(가격·광고비·진열 등)을 추가로 파악할 필요가 있다.
- 최대 판매량은 16,270개로, 우수 지역의 성공 요인 분석이 비즈니스에 도움이 될 수 있다.

5) 추가 분석해 볼 사항이 있나요?

- 판매량 하위 10% 지역과 상위 10% 지역을 구분하여, 어떤 변수가 판매량 차이를 만드는지 이변량 분석으로 연결해 볼 수 있다.
- 판매량이 0에 가까운 지역(이상치 후보)을 별도 필터링하여 공통점 파악

## 3.2. CompPrice

In [ ]:
eda_num(car, 'CompPrice')

1) 변수의 비즈니스 의미

- 경쟁사 가격(단위:달러)

2) 기초통계량(수치화)

In [ ]:
car['CompPrice'].describe()

3) 분포 확인(시각화)

In [ ]:
fig, axes = plt.subplots(1, 2, figsize=(12, 4))

car['CompPrice'].hist(ax=axes[0], bins=20, edgecolor='k', color='steelblue', alpha=0.7)
axes[0].axvline(car['CompPrice'].mean(),   color='red',   linestyle='--', label=f"mean={car['CompPrice'].mean():.2f}")
axes[0].axvline(car['CompPrice'].median(), color='green', linestyle='--', label=f"median={car['CompPrice'].median():.2f}")
axes[0].set_title('CompPrice Histogram')
axes[0].legend()

axes[1].boxplot(car['CompPrice'])
axes[1].set_title('CompPrice Boxplot')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 경쟁사 가격의 평균은 약 **125달러**이며, 분포가 정규분포에 가깝고 이상치가 거의 없다.
- 지역마다 경쟁사 가격이 77~175달러로 약 **2배 이상** 차이 나, 지역별 가격 경쟁 환경이 매우 다르다.
- 경쟁사 가격이 낮은 지역은 가격 경쟁 압박이 크므로, 자사 가격 전략과 연계하여 분석할 필요가 있다.

5) 추가 분석해 볼 사항이 있나요?

- 자사 가격(Price)과 경쟁사 가격의 차이(Price - CompPrice)를 계산하여, 가격 경쟁력을 지역별로 비교해볼 수 있다.
- 경쟁사 가격이 낮은 지역에서의 판매량 변화를 확인하면 가격 탄력성 파악에 도움이 된다.

## 3.3. Income

In [ ]:
eda_num(car, 'Income')

1) 변수의 비즈니스 의미

- 지역 평균 소득(단위: 1000달러)

2) 기초통계량(수치화)

In [ ]:
car['Income'].describe()

3) 분포 확인(시각화)

In [ ]:
car['Income'].hist(bins=20, edgecolor='k', color='steelblue', alpha=0.7, figsize=(8, 4))
plt.axvline(car['Income'].mean(),   color='red',   linestyle='--', label=f"mean={car['Income'].mean():.2f}")
plt.axvline(car['Income'].median(), color='green', linestyle='--', label=f"median={car['Income'].median():.2f}")
plt.title('Income Histogram')
plt.xlabel('Income (1000달러)')
plt.ylabel('Frequency')
plt.legend()
plt.show()

In [ ]:
plt.figure(figsize=(6, 5))
plt.boxplot(car['Income'], vert=True)
plt.title('Income Boxplot')
plt.ylabel('Income (1000달러)')
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 지역 평균 소득은 21~120천 달러로 분포가 넓고, 분포 형태는 비교적 균등(uniform)에 가깝다.
- 평균(68.7천 달러)과 중앙값(69천 달러)이 거의 같아 특정 소득 구간에 쏠림 현상은 없다.
- 소득 수준이 카시트 구매력에 직결될 수 있으므로, 소득 구간별 판매량 차이를 확인할 필요가 있다.

5) 추가 분석해 볼 사항이 있나요?

- 소득 구간(저/중/고)으로 나누어 각 구간별 평균 판매량을 비교해보면 구매력과 판매의 관계를 파악할 수 있다.
- 고소득 지역에서 프리미엄 가격 전략이 효과적인지 Price와 함께 이변량 분석 연결

## 3.4. Advertising

In [ ]:
eda_num(car, 'Advertising')

1) 변수의 비즈니스 의미

- 각 지역, 회사의 광고 예산(단위:1000달러)

2) 기초통계량(수치화)

In [ ]:
car['Advertising'].describe()

3) 분포 확인(시각화)

In [ ]:
car['Advertising'].hist(bins=20, edgecolor='k', color='steelblue', alpha=0.7, figsize=(8, 4))
plt.axvline(car['Advertising'].mean(),   color='red',   linestyle='--', label=f"mean={car['Advertising'].mean():.2f}")
plt.axvline(car['Advertising'].median(), color='green', linestyle='--', label=f"median={car['Advertising'].median():.2f}")
plt.title('Advertising Histogram')
plt.xlabel('Advertising (1000달러)')
plt.ylabel('Frequency')
plt.legend()
plt.show()

In [ ]:
# 광고비 0인 지역 비율
zero_count = (car['Advertising'] == 0).sum()
total = len(car)
print(f'광고비 0인 지역: {zero_count}개 ({zero_count/total*100:.1f}%)')
print(f'광고비 집행 지역: {total - zero_count}개 ({(total - zero_count)/total*100:.1f}%)')

In [ ]:
plt.figure(figsize=(6, 5))
plt.boxplot(car['Advertising'], vert=True)
plt.title('Advertising Boxplot')
plt.ylabel('Advertising (1000달러)')
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 광고 예산의 평균은 약 **6.6천 달러**이나, 최솟값이 0이고 중앙값(5)이 평균보다 낮아 분포가 오른쪽으로 치우쳐 있다(양의 왜도).
- **광고비 0인 지역이 다수** 존재하며, 이 지역들은 광고 없이도 판매가 이루어지는지 확인이 필요하다.
- 광고비를 많이 집행하는 일부 지역(최대 29천 달러)은 이상치에 가까운 수준으로, 효과 대비 예산 집행이 적절한지 검토해야 한다.

5) 추가 분석해 볼 사항이 있나요?

- 광고비 0인 지역과 광고비 집행 지역의 평균 판매량을 비교하면 광고 효과를 간접적으로 확인할 수 있다.
- 광고비 구간별(0, 저, 중, 고)로 나누어 판매량과의 관계를 이변량 분석으로 연결

## 3.5. Population

1) 변수의 비즈니스 의미

- 지역 인구수(단위:1000명)

2) 기초통계량(수치화)

In [ ]:
car['Population'].describe()

3) 분포 확인(시각화)

In [ ]:
fig, axes = plt.subplots(1, 2, figsize=(12, 4))

car['Population'].hist(ax=axes[0], bins=20, edgecolor='k', color='steelblue', alpha=0.7)
axes[0].axvline(car['Population'].mean(),   color='red',   linestyle='--', label=f"mean={car['Population'].mean():.2f}")
axes[0].axvline(car['Population'].median(), color='green', linestyle='--', label=f"median={car['Population'].median():.2f}")
axes[0].set_title('Population Histogram')
axes[0].legend()

axes[1].boxplot(car['Population'])
axes[1].set_title('Population Boxplot')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 지역 인구는 10~509천 명으로 범위가 매우 넓고, 분포가 거의 균등(uniform)에 가깝다.
- 평균(264.8천)과 중앙값(272천)이 유사하여 특정 인구 규모에 쏠림은 없다.
- 인구와 판매량의 관계가 직관적으로는 양의 상관관계가 예상되지만, 실제 데이터에서 어느 정도 연관이 있는지 이변량 분석이 필요하다.

5) 추가 분석해 볼 사항이 있나요?

- 인구 규모를 소/중/대 구간으로 나누어 판매량 평균을 비교해보면 시장 규모 효과를 확인할 수 있다.
- 인구가 많지만 판매량이 낮은 지역이 있다면, 광고·진열 등 다른 요인과 결합하여 원인 파악

## 3.6. Price

1) 변수의 비즈니스 의미

- 자사 지역별 판매가격


2) 기초통계량(수치화)

In [ ]:
car['Price'].describe()

3) 분포 확인(시각화)

In [ ]:
fig, axes = plt.subplots(1, 2, figsize=(12, 4))

car['Price'].hist(ax=axes[0], bins=20, edgecolor='k', color='steelblue', alpha=0.7)
axes[0].axvline(car['Price'].mean(),   color='red',   linestyle='--', label=f"mean={car['Price'].mean():.2f}")
axes[0].axvline(car['Price'].median(), color='green', linestyle='--', label=f"median={car['Price'].median():.2f}")
axes[0].set_title('Price Histogram')
axes[0].legend()

axes[1].boxplot(car['Price'])
axes[1].set_title('Price Boxplot')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 자사 가격 평균은 약 **116달러**이며, 분포가 비교적 대칭적이고 정규분포에 가깝다.
- 가격 범위는 24~191달러로 상당히 넓어, 지역별로 가격 정책이 크게 다름을 알 수 있다.
- 가격이 높을수록 판매량이 줄어드는(음의 관계) 경향이 있을 것으로 예상되며, 이변량 분석으로 검증 필요하다.

5) 추가 분석해 볼 사항이 있나요?

- 경쟁사 가격(CompPrice) 대비 자사 가격 비율을 계산하여 가격 경쟁력 지수를 만들어볼 수 있다.
- 가격이 낮은 지역에서 실제 판매량이 높은지 Sales와의 이변량 분석으로 연결

(추가) Price vs CompPrice

In [ ]:
car[['Price', 'CompPrice']].describe()

In [ ]:
car['PriceDiff'] = car['Price'] - car['CompPrice']
print(car['PriceDiff'].describe())
print(f'\n자사가 비싼 지역: {(car["PriceDiff"] > 0).sum()}개 ({(car["PriceDiff"] > 0).mean()*100:.1f}%)')
print(f'자사가 저렴한 지역: {(car["PriceDiff"] < 0).sum()}개 ({(car["PriceDiff"] < 0).mean()*100:.1f}%)')

fig, axes = plt.subplots(1, 2, figsize=(12, 4))

axes[0].boxplot([car['Price'], car['CompPrice']], labels=['Price(자사)', 'CompPrice(경쟁사)'])
axes[0].set_title('Price vs CompPrice Boxplot')
axes[0].set_ylabel('가격 (달러)')

car['PriceDiff'].hist(ax=axes[1], bins=20, edgecolor='k', color='steelblue', alpha=0.7)
axes[1].axvline(0, color='red', linestyle='--', label='차이=0')
axes[1].set_title('Price - CompPrice 분포')
axes[1].set_xlabel('자사가격 - 경쟁사가격 (달러)')
axes[1].legend()

plt.tight_layout()
plt.show()

car.drop(columns=['PriceDiff'], inplace=True)

## 3.7. Age

1) 변수의 비즈니스 의미

- 지역 인구의 평균 연령

2) 기초통계량(수치화)

In [ ]:
car['Age'].describe()

3) 분포 확인(시각화)

In [ ]:
fig, axes = plt.subplots(1, 2, figsize=(12, 4))

car['Age'].hist(ax=axes[0], bins=20, edgecolor='k', color='steelblue', alpha=0.7)
axes[0].axvline(car['Age'].mean(),   color='red',   linestyle='--', label=f"mean={car['Age'].mean():.2f}")
axes[0].axvline(car['Age'].median(), color='green', linestyle='--', label=f"median={car['Age'].median():.2f}")
axes[0].set_title('Age Histogram')
axes[0].legend()

axes[1].boxplot(car['Age'])
axes[1].set_title('Age Boxplot')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 지역 평균 연령은 25~80세로 분포가 넓고, 평균(53.3세)과 중앙값(54.5세)이 비슷하여 대칭적인 분포이다.
- 분포 모양이 비교적 균등하여 젊은 지역부터 고령 지역까지 고르게 분포한다.
- 카시트는 영유아 자녀를 둔 부모가 주요 고객층이므로, 평균 연령이 낮은(젊은) 지역에서 판매량이 높을 가능성이 있다.

5) 추가 분석해 볼 사항이 있나요?

- 평균 연령을 기준으로 '젊은 지역(40세 미만)'과 '고령 지역(60세 이상)'으로 나누어 판매량 차이를 비교
- 연령과 판매량의 이변량 분석으로 타깃 시장(젊은 인구 비중이 높은 지역) 우선순위 파악

# 4.단변량 분석 : 범주형 변수

In [ ]:
def eda_cat(df, col):
    # 기초통계량
    vc = df[col].value_counts()
    vc_ratio = df[col].value_counts(normalize=True)
    display(pd.DataFrame({'count': vc, 'ratio(%)': vc_ratio * 100}))

    # 시각화
    fig, axes = plt.subplots(1, 2, figsize=(12, 4))

    vc.plot(kind='bar', ax=axes[0], color='steelblue', edgecolor='k', alpha=0.7)
    axes[0].set_title(f'{col} - 빈도')
    axes[0].set_ylabel('Count')
    axes[0].tick_params(axis='x', rotation=0)
    for i, v in enumerate(vc):
        axes[0].text(i, v + 1, str(v), ha='center', va='bottom')

    axes[1].pie(vc, labels=vc.index, autopct='%1.1f%%', startangle=90)
    axes[1].set_title(f'{col} - 비율')

    plt.suptitle(col, fontsize=13, fontweight='bold')
    plt.tight_layout()
    plt.show()

## 4.1. Urban

1) 변수의 비즈니스 의미

- 도심 지역 여부(Yes, No)

2) 기초통계량(수치화)

In [ ]:
car['Urban'].value_counts()

In [ ]:
car['Urban'].value_counts(normalize=True)

In [ ]:
vc = car['Urban'].value_counts()
vc_ratio = car['Urban'].value_counts(normalize=True)
pd.DataFrame({'count': vc, 'ratio(%)': vc_ratio * 100})

3) 분포 확인(시각화)

In [ ]:
vc = car['Urban'].value_counts()

fig, axes = plt.subplots(1, 2, figsize=(12, 4))

vc.plot(kind='bar', ax=axes[0], color=['steelblue', 'orange'], edgecolor='k', alpha=0.8)
axes[0].set_title('Urban 빈도')
axes[0].set_ylabel('Count')
axes[0].tick_params(axis='x', rotation=0)
for i, v in enumerate(vc):
    axes[0].text(i, v + 1, str(v), ha='center', va='bottom')

axes[1].pie(vc, labels=vc.index, autopct='%1.1f%%', startangle=90, colors=['steelblue', 'orange'])
axes[1].set_title('Urban 비율')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 전체 400개 지역 중 약 **72%가 도심 지역**(Urban=Yes)으로, 대부분의 매장이 도시에 위치한다.
- 비도심 지역(28%)도 적지 않으므로, 도심 vs 비도심 간 판매량 차이를 확인할 필요가 있다.
- 도심 지역의 인구 밀도나 소득 수준 차이가 판매량에 영향을 줄 수 있다.

5) 추가 분석해 볼 사항이 있나요?

- 도심(Yes)/비도심(No) 그룹별 평균 판매량 비교로 도심 입지 효과 검증
- Urban + US 교차 분석으로 미국 내 도심 매장과 해외 도심 매장의 차이 파악

## 4.2. ShelveLoc

1) 변수의 비즈니스 의미

* 매장 진열상태

2) 기초통계량(수치화)

In [ ]:
car['ShelveLoc'].value_counts()

In [ ]:
vc = car['ShelveLoc'].value_counts()
vc_ratio = car['ShelveLoc'].value_counts(normalize=True)
pd.DataFrame({'count': vc, 'ratio(%)': vc_ratio * 100})

3) 분포 확인(시각화)

In [ ]:
vc = car['ShelveLoc'].value_counts()

fig, axes = plt.subplots(1, 2, figsize=(12, 4))

vc.plot(kind='bar', ax=axes[0], color=['steelblue', 'orange', 'green'], edgecolor='k', alpha=0.8)
axes[0].set_title('ShelveLoc 빈도')
axes[0].set_ylabel('Count')
axes[0].tick_params(axis='x', rotation=0)
for i, v in enumerate(vc):
    axes[0].text(i, v + 1, str(v), ha='center', va='bottom')

axes[1].pie(vc, labels=vc.index, autopct='%1.1f%%', startangle=90)
axes[1].set_title('ShelveLoc 비율')

plt.tight_layout()
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 진열 상태는 Bad(96개, 24%), Good(85개, 21.3%), Medium(219개, 54.8%)로 **Medium이 절반 이상**을 차지한다.
- Good 진열 매장이 전체의 21%에 불과하여, 진열 개선을 통한 판매량 증대 여지가 있다.
- 진열 상태는 직접 관리 가능한 변수이므로, 판매량과의 관계를 확인하면 즉각적인 실행 방안을 도출할 수 있다.

5) 추가 분석해 볼 사항이 있나요?

- 진열 상태(Bad/Medium/Good)별 평균 판매량 비교: 좋은 진열이 실제로 매출에 기여하는지 이변량 분석으로 검증
- Bad 진열 매장을 Medium 이상으로 개선했을 때 기대 매출 증가량 추정

## 4.3. US

In [ ]:
eda_cat(car, 'US')

1) 변수의 비즈니스 의미

- 매장이 미국에 있는지 여부(Yes, No)

2) 기초통계량(수치화)

In [ ]:
car['US'].value_counts()

In [ ]:
vc = car['US'].value_counts()
vc_ratio = car['US'].value_counts(normalize=True)
pd.DataFrame({'count': vc, 'ratio(%)': vc_ratio * 100})

3) 분포 확인(시각화)

In [ ]:
vc = car['US'].value_counts()

vc.plot(kind='bar', color=['steelblue', 'orange'], edgecolor='k', alpha=0.8, figsize=(6, 4))
plt.title('US 빈도')
plt.ylabel('Count')
plt.xticks(rotation=0)
for i, v in enumerate(vc):
    plt.text(i, v + 1, str(v), ha='center', va='bottom')
plt.tight_layout()
plt.show()

In [ ]:
vc = car['US'].value_counts()

plt.figure(figsize=(5, 5))
plt.pie(vc, labels=vc.index, autopct='%1.1f%%', startangle=90, colors=['steelblue', 'orange'])
plt.title('US 매장 비율')
plt.show()

4) 기초통계량과 분포를 통해서 파악한 내용을 적어 봅시다.

보이는 그대로를 넘어, 비즈니스 관점에서 고민하며 적어 봅시다.

- 전체 매장 중 약 **70%가 미국 내 매장**(US=Yes), 30%는 해외 매장이다.
- 해외 매장이 30%로 적지 않은 비중을 차지하나, 부서의 '국내 시장 편중' 우려가 있으므로 해외 매장의 판매 성과를 확인해야 한다.
- 미국(US=Yes)과 해외(US=No) 매장 간 판매량 차이를 분석하여 해외 시장 가능성을 평가할 필요가 있다.

5) 추가 분석해 볼 사항이 있나요?

- US=Yes vs US=No 그룹별 평균 판매량, 가격, 광고비를 비교하여 해외 매장 특성 파악
- 해외 매장 중 판매량이 높은 지역의 공통 특성(진열, 가격, 광고 등) 추출로 해외 확장 전략 수립